ethereum/contracts/Bridge.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.2;
+
+import "./Decoder.sol";
+import "./Application.sol";
+
+contract Bridge {
+    using Decoder for bytes;
+
+    uint64 public mostRecentBlock;
+    mapping(uint64 => uint64) public mostRecentBlockEvent;
+    mapping(address => bool) public applications;
+
+
+    constructor(address[] memory apps) public {
+        mostRecentBlock = 0;
+        for(uint256 i = 0; i < apps.length; i++) {
+            if(verifyApp(apps[i])) {
+                applications[apps[i]] = true;
+            }
+        }
+    }
+
+    /**
+     * @dev routes the message to the specified application ID after verifying the operator's signature
+     * @param _message address _message expected type: Message { AppID [32]byte, Payload []byte }
+     * @param _verification bytes _verification expected type: Verification { BlockNumber int64, EventID int64 }
+     */
+    function handle(bytes memory _message, bytes memory _verification)
+        public
+    {
+        uint64 blockNumber = uint64(_verification.slice(0, 8).decodeUint256());
+        require(blockNumber >= mostRecentBlock, "Blocks must be processed chronologically");
```

Comment:
Surely you mean `blockNumber < mostRecentBlock`?

Reply:
Here `mostRecentBlock` refers to the block associated with the most recently processed event. If we've already processed event 1 from block 100, we shouldn't process any events from block 1-99. We only want to process additional events from block 100 or blocks greater than 100.